lab/lablib/SoundManager.js

```diff
@@ -199,6 +239,9 @@ export class SoundManager {
   melody_b() {
```

Comment:
Try writing a third melody that involves taking a motif and transforming the pitches. It would be helpful to add some functions on `Music<P> -> Music<P>` such as:

- `transpose(interval, music)` - increment the pitches by the given interval
- `retrograde(music)` - reverse everything in the music
- `invert(center, music)` - Invert the pitches around a center